proxy/Main.cc

```diff
@@ -476,6 +476,18 @@ check_config_directories(void)
   }
 }
 
+namespace {
+  // Values with names that have this prefix are exempted from being considered "unrecognized".
+  const char PLUGIN_CONFIG_RECORD_PREFIX[] = "proxy.config.plugin.";
```

Comment:
I think the right check is to test whether there are any unregistered local or config records. This can be done periodically since new records can be added at config reload time. A records that is present in ``records.config`` will be not be marked as registered intul a plugin calls ``TSMgmtStringCreate()`` or the equivalent.